agimus_controller/agimus_controller/factory/robot_model.py

```diff
@@ -1,96 +1,219 @@
+        """Load the reduced robot model."""
+        joints_to_lock = self._get_joints_to_lock()
+        self._robot_model = pin.buildReducedModel(
+            self._full_robot_model, joints_to_lock, self._q0
         )
 
-    def _load_pinocchio_models(self, urdf: Path, free_flyer: bool) -> None:
-        pass
-
-    def _load_default_configuration(self, srdf_path: Path, q0_name: str) -> None:
-        pass
-
-    def _load_reduced_model(self, locked_joint_names, q0_name) -> None:
-        pass
-
-    def _update_collision_model(
-        self,
-        env: Union[Path, None],
-        collision_as_capsule: bool,
-        self_collision: bool,
-        srdf: Path,
-    ) -> None:
-        pass
-
-    def create_complete_robot_model(self) -> pin.Model:
-        return self._complete_model.copy()
-
-    def create_complete_collision_model(self) -> pin.GeometryModel:
-        return self._complete_collision_model.copy()
-
-    def create_complete_visual_model(self) -> pin.GeometryModel:
-        return self._complete_visual_model.copy()
-
-    def create_reduced_robot_model(self) -> pin.Model:
-        return self._rmodel.copy()
-
-    def create_reduced_collision_model(self) -> pin.GeometryModel:
-        return self._rcmodel.copy()
-
-    def create_reduced_visual_model(self) -> pin.GeometryModel:
-        return self._rvmodel.copy()
-
-    def create_default_configuration(self) -> np.array:
-        return self._q0.copy()
-
-    def create_model_parameters(self) -> RobotModelParameters:
-        return deepcopy(self._params)
+    def _update_collision_model_to_capsules(self) -> None:
+        """Update the collision model to capsules."""
+        cmodel = self._collision_model.copy()
+        list_names_capsules = []
+        # Iterate through geometry objects in the collision model
+        for geom_object in cmodel.geometryObjects:
+            geometry = geom_object.geometry
+            # Remove superfluous suffix from the name
+            base_name = "_".join(geom_object.name.split("_")[:-1])
+            # Convert cylinders to capsules
+            if isinstance(geometry, coal.Cylinder):
+                name = self._generate_capsule_name(base_name, list_names_capsules)
+                list_names_capsules.append(name)
+                capsule = pin.GeometryObject(
+                    name=name,
+                    parent_frame=int(geom_object.parentFrame),
+                    parent_joint=int(geom_object.parentJoint),
+                    collision_geometry=coal.Capsule(
+                        geometry.radius, geometry.halfLength
+                    ),
+                    placement=geom_object.placement,
+                )
+                capsule.meshColor = np.array([249, 136, 126, 125]) / 255  # Red color
+                self._collision_model.addGeometryObject(capsule)
+                self._collision_model.removeGeometryObject(geom_object.name)
+
+            # Remove spheres associated with links
```

Comment:
What is the rationale for removing spheres ?

Reply:
Relying on the name can be rather unexpected to users so I would prefer not to do so.

If you define a capsule as 2 spheres and one cylinder in the URDF, then, I see two solutions:
- rely on the way Pinocchio URDF parser names geometries. You will have 3 geoms with name `link_name_[0_9]`. Then, you have to check whether the 3 are 2 spheres and 1 cylinder. 
- a more robust method would be :
  - for each joint:
    - look for all 3-ple (cylinder, sphere, sphere) attached to that joint
    - for each 3-ple:
      - if the two sphere have the same radius as the cylinder and the center of the spheres matches the center of the two cylinder extreme circles, then replace them by a capsule.

The second is a little more complex to implement but is clearly the most robust.

Reply:
I created #117

Reply:
We can leave this for a future work.